### PR TITLE
Added tests and notebook demo for MSA level

### PR DESCRIPTION
**Pull request by Niklewa**
I have modified tests for functions that were generalized on separate branches to incorporate MSA-level variables. In addition, I have added a notebook that tests the functionalities of the `MSAFipsQuery` class.

### CODE REVIEW

File: cities/queries/fips_query.py [hunk 1]
```diff
@@ -74,26 +74,28 @@ def __init__(
         self.fips = fips
         self.lag = lag
         self.top = top
+        self.gdpVAR = "gdp"
```
Comment (rfl-urbaniak) [inline cities/queries/fips_query.py:77]: can you explain why use `self.gdpVAR` instead of a shorter string
`gdp`? Also, if you have a good reason and insist, rename as `self.gdp_var`.

Comment (rfl-urbaniak) [reply cities/queries/fips_query.py:77]: Ah, ok, I see your reason, you want this to apply to `gdp_ma` in the subclass. Still, can you rename please?:)

File: cities/queries/fips_query.py [hunk 2]
```diff
@@ -631,16 +646,101 @@ def __init__(
         outcome_comparison_period=None,
         outcome_percentile_range=None,
     ):
-        super().__init__(
```
Comment (rfl-urbaniak) [inline cities/queries/fips_query.py:634]: So you seem to be repeating most of the __init__, except perhaps for

self.data = MSADataGrabber()
self.gdpVAR = "gdp_ma"

(anything else?)

Did you try calling super init and overwriting instead? Or some other way to avoid repeating so much code?

File: tests/test_data_grabber.py [hunk 3]
```diff
@@ -135,3 +135,25 @@ def test_feature_listing_runtime():
...
+
+
+feature_ma = list_available_features('msa')
+data = MSADataGrabber()
+data.get_features_long(feature_ma)
+
+def test_GeoFIPS_column_values():
+    for feature in feature_ma:
+
+        data.long[feature]['GeoFIPS']
```
Comment (rfl-urbaniak) [inline tests/test_data_grabber.py]: why is this here?

File: tests/test_data_grabber.py [hunk 4]
```diff
@@ -135,3 +135,25 @@ def test_feature_listing_runtime():
...
+def test_ma_strings_in_features():
+    feature_ma = list_available_features('msa')
+    assert all(feature.endswith("_ma") for feature in feature_ma)
+
+
+feature_ma = list_available_features('msa')
+data = MSADataGrabber()
+data.get_features_long(feature_ma)
+
+def test_GeoFIPS_column_values():
```
Comment (rfl-urbaniak) [inline tests/test_data_grabber.py:156]: the name doesn't specify that it's an MSA test.

File: tests/test_data_grabber.py [hunk 5]
```diff
@@ -135,3 +135,25 @@ def test_feature_listing_runtime():
...
+    features = list_available_features()
+    assert all(not feature.endswith("_ma") for feature in features)
+
+def test_ma_strings_in_features():
+    feature_ma = list_available_features('msa')
+    assert all(feature.endswith("_ma") for feature in feature_ma)
+
+
+feature_ma = list_available_features('msa')
+data = MSADataGrabber()
```
Comment (rfl-urbaniak) [inline tests/test_data_grabber.py:152]: Most of the tests performed on `DataGrabber` in `test_DataGrabber()` are now not run on `MSADataGrabber`. Refactor this test by parametrizing it by which data grabber is used and applying all checks that make sense to the latter as well (and the other ones conditionally if dealing with the former).

**Review by rfl-urbaniak — CHANGES_REQUESTED** [review]
Thank you for your work on this. The functionality seems more or less ready, some refactoring, a few nits, and a few changes requested when it comes to testing.

File: tests/test_data_grabber.py (hunk 5 above)
Comment (Niklewa) [reply tests/test_data_grabber.py]: done

**Review by Niklewa — COMMENTED** [review]
I have addressed all the issues

**Review by rfl-urbaniak — CHANGES_REQUESTED** [review]
fails make lint fails make test

**Review by rfl-urbaniak — CHANGES_REQUESTED** [review]
add query parametrized test of kins don't die for the msa level please

**Comment by Niklewa** [comment]
> add query parametrized test of kins don't die for the msa level please

done

**Review by rfl-urbaniak — APPROVED** [review]
